core/src/main/java/org/pentaho/platform/util/ActionUtil.java

```diff
@@ -503,10 +503,10 @@ private static void addAttachment( Map<String, Object> actionParams, Map<String,
       emailer.setAttachmentName( attachmentName.endsWith( extension ) ?  attachmentName : attachmentName + extension );
     } else if ( data != null ) {
       String path = filePath;
-      if ( path.endsWith( ".*" ) ) {
-        path = path.replace( ".*", "" );
-      }
       String extension = MimeHelper.getExtension( data.getMimeType(), ".bin" );
+      if ( path.endsWith( extension ) ) {
+        path = path.replace( extension, "" );
```

Comment:
Safer to use substring. Something like `path.substring(0, path.length() - extension.length() )`.

Reply:
Ok

Reply:
I had just replaced the regex (which was not correct) with the actual value of extension.

Reply:
done